src/main/java/com/atlassian/db/replica/internal/SQLOpeartionType.java

```diff
@@ -0,0 +1,50 @@
+package com.atlassian.db.replica.internal;
+
+public class SQLOpeartionType {
```

Comment:
The OOP way:
```java
public class SqlQuery {

    private final String query;
...
    public Boolean writesToDatabase() {
```

Reply:
typo: SQL**Opeartion**Type

Reply:
I like the idea.

SqlQuery.writesToDatabase() I think to answer this not only query is needed, but also the `context`

What's the flow /source of the method call it (I mean `execute`,`executeQuery`, `executeUpdate` etc.. 
If you think about expanding it even more-> isReplicaConsisten might be required too.

This make this complex, don't you think so ?
Or maybe you have better idea to implement it ?

Reply:
It's just a query, no context

Reply:
I havent introduce SQLQuery, it wasnt required this time. But it's worth to remember, maybe next time there will be more reasons to push it.

Reply:
"Wasn't required"? I'd say `SQLOperationType` isn't required.

Reply:
True,` SQLOperationType` isn't required too. But i think it's natural extraction of all methods with similar responsibility to separate class.

Reply:
As with `SqlQuery` this brings more OOP while currently `String sql` is used widely in whole db-replica. 
I don't see a reason to extract it to a separate class everyhwere, especially as `isWrite` methods are used only inside `ReplicaStatement` and in their context. 
Why in their context?
We can say whether SqlQuery is fully read, as this is recognized on ReplicaStatement, and depends from which method was called.

Maybe you had different idea for the use of `SQL query` ?  I'm open for suggestion.

Reply:
I've changed `SQLOperationType` it od  `SqlFunctionChecker`

Reply:
`*Checker`... 🤔 . Will it trigger @dagguh? 🍿  😄

Reply:
I don't understand the defense of procedural programming.

Reply:
> defense of procedural programming

Ok those words convinced me.

Reply:
`SqlQuery` introduced, Friday afternoon wasn't productive, class name with verbs aren't good idea. 🤦